source/FreeRTOS_ND.c

```diff
@@ -639,9 +639,18 @@
                 ( void ) memcpy( pxICMPHeader_IPv6->ucOptionBytes, pxEndPoint->xMACAddress.ucBytes, ipMAC_ADDRESS_LENGTH_BYTES );
 
                 /* Checksums. */
-                pxICMPHeader_IPv6->usChecksum = 0U;
-                /* calculate the ICMP checksum for the outgoing package. */
-                ( void ) usGenerateProtocolChecksum( pxDescriptor->pucEthernetBuffer, pxDescriptor->xDataLength, pdTRUE );
+                #if ( ipconfigDRIVER_INCLUDED_TX_IP_CHECKSUM == 0 )
+                    {
+                        /* calculate the ICMPv6 checksum for outgoing package */
```

Comment:
Is the pxICMPHeader_IPv6->usChecksum = 0U; not needed here? Earlier we were calling generate checksum after initializing to zero.

Reply:
It's not necessary because we use 0 as initial value when calculating checksum.